Mac-App/Backend/Backend/Shared/Helpers /ConcurrentWork.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ConcurrentWork.swift
+//  Backend
+//
+//  Created by Daniel Garcia on 19.06.20.
+//  Copyright © 2020 Acphut Werkstatt. All rights reserved.
+//
+
+import Foundation
+
+func splitAndExecuteConcurrently<I: Hashable, O: Hashable>(
+    _ collection: [I],
+    transform: @escaping (I) -> O?
+) -> [O] {
+    let middleIndex: Double = floor(Double(collection.count / 2))
+    let firstChunk = collection.prefix(upTo: Int(middleIndex))
+    let secondChunk = collection.suffix(from: Int(middleIndex))
+
+    var allItems: Set<O> = []
+    var firstChunkItems: Set<O> = []
+    var secondChunkItems: Set<O> = []
+
+    let operation = OperationQueue()
```

Comment:
I think renaming this to `operationQueue` or `queue` would be better.

Reply:
True

Reply:
https://github.com/Thurman1776-/Uncomplect-us/pull/16/commits/365e4e4c500db1c463c117493edb21cb9688c8f9